File: app/src/main/java/mchehab/com/twowaydatabinding/MainActivity.java
package mchehab.com.twowaydatabinding;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import mchehab.com.twowaydatabinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private Person person = new Person();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setPerson(person);
        binding.setMainActivity(this);
    }

    public void onButtonClick(View view){
        person.setFirstName("first name");
        person.setLastName("last name");
        person.setAge(23);
    }
}

File: kotlinapp/src/main/java/mchehab/com/kotlinapp/MainActivity.kt
package mchehab.com.kotlinapp

import android.databinding.DataBindingUtil
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import mchehab.com.kotlinapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private val person = Person()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        binding.person = person
        binding.mainActivity = this
        binding.converter = Converter()
    }

    fun onButtonClick(view: View) {
        person.firstName = "first name"
        person.lastName = "last name"
        person.age = 23
    }
}
